### PR TITLE
#181339530: log out functionalities

### PR DESCRIPTION
**Pull request by blandri**
### What does this PR do?

this PR is for logging out functionalities

### Description of Task to be completed?
 
- A user logs out using the access token given on log in
- The user is stored in redis database as they log out
- A user cant use the access token when they are logged out

### How should this be manually tested?

- Clone this [repo](https://github.com/atlp-rwanda/cabal-bn-be.git)
- Download redis server on your pc if you are going to test locally

- register on the app
- log in and take the token
- use it to logout

- use postman and login and get the token
         - POST    **http://localhost:3000/api/v1/users/login**
- use postman to logout
         - POST   **http://localhost:3000/api/v1/users/logout**

- Link for swagger: [swagger-logout](http://localhost:3000/api-docs/#/Users)

         

### What are the relevant pivotal tracker stories?

[#181339530](https://www.pivotaltracker.com/n/projects/2555280)

### Screenshots (if appropriate)
image

### CODE REVIEW

**Review by Gabin-ishimwe — CHANGES_REQUESTED** [review]
resolve your circle ci problem for your tests to run

File: tests/user.test.js [hunk 1]
```diff
@@ -109,4 +110,34 @@ describe('USER END-POINT TEST', () => {
...
+
+      const res = await chai
+        .request(app)
+        .post(`/api/v1/users/logout`)
+        .set({ Authorization: `Bearer ${token}` });
+
+      expect(res.status).to.be.equal(401);
+    });
+  });
 });
```
Comment (Eli250) [inline tests/user.test.js:157]: Add new line at the end

File: src/middlewares/authentication.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    if (user) {
+      return res.status(401).json({ message: 'please log in first' });
+    }
+    await jwt.verify(token, process.env.SECRETE);
+    next();
+  } catch (error) {
+    res.status(401).send({ message: 'You are not logged in' });
+  }
+};
```
Comment (Eli250) [inline src/middlewares/authentication.js:24]: Add a new line a the end.

File: src/redis/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import redis from 'redis';
+import { createClient } from '@node-redis/client';
+
+const client = createClient();
+client.on('connect', () => {
+  console.log('Client connected');
+});
+client.connect();
+
+export default client;
```
Comment (Eli250) [inline src/redis/index.js:14]: Add new line at the end.

**Review by Eli250 — CHANGES_REQUESTED** [review]
Add a new line at the end of each file.

**Review by Code-250 — APPROVED** [review]

**Pull request closed without merging**